### PR TITLE
Updated comments and README for the deployment automation

### PR DESCRIPTION
**Pull request by pabowers**
Please give me feedback on things that are unclear so that I may further clarify.  Thanks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 
 2. HANA high-availability pair:
-   To create the infrastructure for the HA pair, terraform will be run from the `deploy/vm/modules/ha_pair` directory.  This will allow us to have new modules for each configuration of the HANA database. Currently, both of the databases, `db0` and `db1` have HANA installed. The `terraform.tfvars` files with the required configuration needs to be put in this folder. An example tfvars file can be found below.
+   To create the infrastructure for the HA pair, terraform will be run from the `deploy/vm/modules/ha_pair` directory.  This will allow us to have new modules for each configuration of the HANA database. Currently, both of the databases, `db0` and `db1` have HANA installed with HSR and failover capabilities. The `terraform.tfvars` files with the required configuration needs to be put in this folder. An example tfvars file can be found below.
 
-Example terraform.tfvars:
+Getting Started
 -------------------------
+You will need to have the following installed on your machine:
```
Comment (lagalbra) [inline README.md]: Can you include links to public documentation on how to install these?

File: README.md [hunk 2]
```diff
@@ -42,3 +49,7 @@ Example terraform.tfvars:
  pw_db_system =  
  useHana2 =  
  install_xsa =  
+
+Additional steps
+-------------------------------------
+You will need to get the urls from SAP as we are not licensed to provide these bits.
```
Comment (lagalbra) [inline README.md]: Be more explicit; I'm assuming by "the urls from SAP", you mean download links for SAP products, but I could be wrong.

File: deploy/vm/modules/create_db_node/main.tf [hunk 3]
```diff
@@ -1,3 +1,4 @@
+# This calls the module to create this node's network interface card and public ip
```
Comment (lagalbra) [inline deploy/vm/modules/create_db_node/main.tf]: nit: capitalize all acronyms used in comments and documentation: "IP", "VM", etc.

File: deploy/vm/modules/create_db_node/variables.tf [hunk 4]
```diff
@@ -1,63 +1,63 @@
+variable "availability_set_id" {
+  description = "The if associated with the availability set to put this vm into."
+  default     = ""                                                                 # empty string denotes that this VM is not in a
```
Comment (lagalbra) [inline deploy/vm/modules/create_db_node/variables.tf]: nit: why such a large space between comment and code on this line? Is this an automatic Terraform linting thing? (if so, that's fine)

Comment (lagalbra) [reply deploy/vm/modules/create_db_node/variables.tf]: "NSG", "NIC"...

File: deploy/vm/modules/ha_pair/variables.tf [hunk 5]
```diff
@@ -3,81 +3,74 @@ variable "ansible_playbook_path" {
...
-  default     = ""
+variable "private_ip_address_lb_frontend" {
+  default = "10.0.0.13"
 }
 
-variable "url_di_core" {
-  description = "URL for DI Core"
-  default     = ""
+variable "public_ip_allocation_type" {
+  description = "Defines whether the IP address is static or dynamic. Options are Static or Dynamic."
```
Comment (lagalbra) [inline deploy/vm/modules/ha_pair/variables.tf]: You say "Options are Static or Dynamic", but the default is "dynamic", lowercase. I would state it with the intended casing, and quotes, like this: `Options are 'static' or 'dynamic'`

File: deploy/vm/modules/ha_pair/variables.tf [hunk 6]
```diff
@@ -86,82 +79,87 @@ variable "pwd_db_tenant" {
...
+variable "url_xs_services" {
+  description = "URL for XS Services"
 }
 
-variable "storage_disk_sizes_gb" {
-  description = "List disk sizes in GB for all disks this VM will need"
-  default     = [512, 512, 512]
+variable "url_xsa_runtime" {
+  description = "URL for XSA runtime"
+  default     = ""
```
Comment (lagalbra) [inline deploy/vm/modules/ha_pair/variables.tf]: Why is there a default value declared for this URl variable type, but not others? Is it always required to be declared, and you do not want to check for null, or something similar?

File: deploy/vm/modules/nsg_for_hana/nsg.tf [hunk 7]
```diff
@@ -1,3 +1,4 @@
+# This gets the ip address of the machine you deploy fromß
```
Comment (lagalbra) [inline deploy/vm/modules/nsg_for_hana/nsg.tf]: What is this weird bobby-pin-shaped character at the end of this line??

File: deploy/vm/modules/nsg_for_hana/nsg.tf [hunk 8]
```diff
@@ -45,12 +48,13 @@ resource "azurerm_network_security_group" "sap-nsg" {
...
   tags {
-    environment = "Terraform SAP HANA single node deployment nsg"
+    environment = "Terraform SAP HANA deployment nsg"
   }
 }
 
+# The Ports that HANA 1 uses are different from the ones HANA 2 uses
 resource "azurerm_network_security_rule" "hana1-http" {
-  count                       = "${!var.useHana2 ? 1 : 0}"
+  count                       = "${!var.useHana2 ? 1 : 0}"                       # The rule is only created if we use HANA 1
```
Comment (lagalbra) [inline deploy/vm/modules/nsg_for_hana/nsg.tf]: Same comment-spacing comment; why so far out?

File: deploy/vm/modules/nsg_for_hana/nsg.tf [hunk 9]
```diff
@@ -65,7 +69,7 @@ resource "azurerm_network_security_rule" "hana1-http" {
 }
 
 resource "azurerm_network_security_rule" "hana1-https" {
-  count                       = "${!var.useHana2 ? 1 : 0}"
+  count                       = "${!var.useHana2 ? 1 : 0}"                       # The rule is only created if we use HANA 1
```
Comment (lagalbra) [inline deploy/vm/modules/nsg_for_hana/nsg.tf]: Same comment-spacing comment; why so far out?

File: deploy/vm/modules/create_db_node/variables.tf (hunk 4 above)
Comment (lagalbra) [reply deploy/vm/modules/create_db_node/variables.tf]: "URL", "SAP", "SSH"...

File: README.md (hunk 2 above)
Comment (pabowers) [reply README.md]: I need to check with Tobias on how to provide instructions for this part since he just gave me the urls

File: deploy/vm/modules/nsg_for_hana/nsg.tf (hunk 7 above)
Comment (pabowers) [reply deploy/vm/modules/nsg_for_hana/nsg.tf]: complete typo. I'm not even sure how I did that.

File: deploy/vm/modules/create_db_node/main.tf [hunk 10]
```diff
@@ -11,6 +12,7 @@ module "nic_and_pip_setup" {
   backend_ip_pool_ids       = "${var.backend_ip_pool_ids}"
 }
 
+# This module creates a vm and the disks that it will need
```
Comment (persiaAziz) [inline deploy/vm/modules/create_db_node/main.tf]: Can 'it' be replaced by HANA db or something?

File: deploy/vm/modules/ha_pair/variables.tf (hunk 6 above)
Comment (persiaAziz) [reply deploy/vm/modules/ha_pair/variables.tf]: This variable is only required when installing XSA, otherwise terraform asks the user to enter this. Having a default avoids that.

Comment (lagalbra) [reply deploy/vm/modules/ha_pair/variables.tf]: ack

File: deploy/vm/modules/ha_pair/variables.tf (hunk 5 above)
Comment (pabowers) [reply deploy/vm/modules/ha_pair/variables.tf]: I'll fix mine, but terraform did it wrong in the documentation.  https://www.terraform.io/docs/providers/azurerm/r/public_ip.html#public_ip_address_allocation 
My bad in copying that comment

Comment (pabowers) [reply deploy/vm/modules/ha_pair/variables.tf]: nvm, I misread

File: deploy/vm/modules/create_db_node/variables.tf (hunk 4 above)
Comment (pabowers) [reply deploy/vm/modules/create_db_node/variables.tf]: yes the spacing is a terraform linting thing.  I tried to move them closer, but then ran `terraform fmt`, which is how to format the Terraform files, and the space just got added again.

File: deploy/vm/modules/nsg_for_hana/nsg.tf (hunk 8 above)
Comment (pabowers) [reply deploy/vm/modules/nsg_for_hana/nsg.tf]: terraform linting thing

File: deploy/vm/modules/nsg_for_hana/nsg.tf (hunk 9 above)
Comment (pabowers) [reply deploy/vm/modules/nsg_for_hana/nsg.tf]: same as above

Comment (lagalbra) [reply deploy/vm/modules/nsg_for_hana/nsg.tf]: ack

File: README.md (hunk 2 above)
Comment (abhinababasu) [reply README.md]: I understand we don't have this fully figured out. This shouldn't block this PR because we will not be able to close the URLs yet.
I would say word it like 
Please download the relevant SAP installers and store them in accessible blobs, and use SAS urls to those blobs for installation. 
Give more info to do that if required

File: deploy/vm/modules/generic_nic_and_pip/outputs.tf [hunk 11]
```diff
@@ -1,7 +1,9 @@
 output "fqdn" {
-  value = "${azurerm_public_ip.pip.fqdn}"
+  description = "The fully qualified domain name associated with the public ip that was created."
+  value       = "${azurerm_public_ip.pip.fqdn}"
 }
 
 output "nic_id" {
-  value = "${azurerm_network_interface.nic.id}"
+  description = "The id of the network interface card will be needed to attach it to the vm."
```
Comment (lagalbra) [inline deploy/vm/modules/generic_nic_and_pip/outputs.tf]: nit: VM

File: README.md [hunk 12]
```diff
@@ -1,18 +1,25 @@
...
+You will need to have the following installed on your machine:
+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest)
+- [Terraform](https://www.terraform.io/intro/getting-started/install.html)
+- [Ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#latest-releases-via-pip)
+
 
- #Example tfvars file  
+Example terraform.tfvars:
+-------------------------
  az_region =
```
Comment (lagalbra) [inline README.md]: When looking at the resulting markdown, it looks a little odd to have an `=` symbol followed by nothing; since it's just example variables now, instead of an example file, I would recommend removing everything except the variable names.

File: deploy/vm/modules/generic_nic_and_pip/outputs.tf (hunk 11 above)
Comment (pabowers) [reply deploy/vm/modules/generic_nic_and_pip/outputs.tf]: That sneaky comment got past me.  I thought I got all of them

File: README.md (hunk 12 above)
Comment (pabowers) [reply README.md]: That wasn't my change, but it does show the syntax.  I vote to keep it there so that the user doesn't use a colon instead.

Comment (lagalbra) [reply README.md]: So it is still supposed to demonstrate the file syntax? In that case, I would reword the heading slightly to "Example terraform.tfvars file"
